File: dandanplay/lib/Model/Message/Receive/ParseFileMessage.dart

import 'package:dandanplay/Model/BaseModel.dart';
import 'package:dandanplay/Protocol/FileMatchProtocol.dart';
import 'package:json_annotation/json_annotation.dart';

part 'ParseFileMessage.g.dart';

@JsonSerializable()
class ParseFileMessage extends BaseModel with FileMatchProtocol {
  String fileName;
  String fileHash;
  num fileSize;
  String mediaId;

  String get matchFileName {
    return fileName;
  }

  Future<String> get matchFileHash async {
    return fileHash;
  }

  Future<int> get matchFileSize async {
    return fileSize;
  }

  ParseFileMessage(this.fileName, this.fileHash, this.fileSize, this.mediaId);

  factory ParseFileMessage.fromJson(Map<String, dynamic> json) {
    return _$ParseFileMessageFromJson(json);
  }

  Map<String, dynamic> toJson() {
    return _$ParseFileMessageToJson(this);
  }
}

File: CocoaShare/Model/Message/Send/ParseFileMessage.swift
//
//  ParseFileMessage.swift
//  dandanplay_native
//
//  Created by JimHuang on 2020/2/16.
//

import Foundation

public class ParseFileMessage: BaseModel, MessageProtocol {
    public var fileName: String?
    public var fileHash: String?
    public var fileSize = 0
    public var mediaId = ""
}
